### PR TITLE
v1.0.0-rc1 release candidate

### PR DESCRIPTION
**Pull request by ehdsouza**
This PR supports the pre-release candidate before the official release in end of september

It includes:
* The new auth mechanism
* The adapter for stt recognize using web socket has been updated to use a temporary request object so that it can authenticate
* Disabled semantic release

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 [[constraint]]
   name = "github.com/IBM/go-sdk-core"
-  version = "0.4.2"
+  version = "0.5.0"
```
Comment (mediumTaj) [inline Gopkg.toml]: no 1.0.0 yet? ;)

File: README.md [hunk 2]
```diff
@@ -111,24 +111,13 @@ You supply either an IAM service **API key** or an **access token**:
...
-
-```go
-// In the constructor, assuming control of managing IAM token
-import "github.com/watson-developer-cloud/go-sdk/discoveryv1"
-
-discovery, discoveryErr := discoveryv1.NewDiscoveryV1(&discoveryv1.DiscoveryV1Options{
-		URL:            "<service_url>",
-		Version:        "2018-03-05",
-		IAMAccessToken: "<IAM_access_token>",
+		Authenticator: authenticator,
```
Comment (mediumTaj) [inline README.md]: extra comma? or are these ok in go? im guessing they're ok because I see them around

File: README.md [hunk 3]
```diff
@@ -275,7 +270,9 @@ func main() {
   discoveryService, discoveryServiceErr := discovery.NewDiscoveryV1(&discovery.DiscoveryV1Options{
     URL:       "YOUR SERVICE URL",
     Version:   "2018-03-05",
-    IAMApiKey: "YOUR IAM API KEY",
+	Authenticator: &core.IamAuthenticator{
+		Apikey: "YOUR APIKEY",
+	},
```
Comment (mediumTaj) [inline README.md]: is indentation off here? i see it in a few examples

File: assistantv2/assistant_v2_test.go [hunk 4]
```diff
@@ -15,31 +31,32 @@ var _ = Describe("AssistantV2", func() {
 	Describe("CreateSession(createSessionOptions *CreateSessionOptions)", func() {
 		createSessionPath := "/v2/assistants/{assistant_id}/sessions"
 		version := "exampleString"
+		accessToken := "0ui9876453"
```
Comment (mediumTaj) [inline assistantv2/assistant_v2_test.go]: we should call these `bearerToken`

File: README.md (hunk 2 above)
Comment (ehdsouza) [reply README.md]: no, its ok

File: assistantv2/assistant_v2_test.go (hunk 4 above)
Comment (ehdsouza) [reply assistantv2/assistant_v2_test.go]: This is something we will update with the automated unit tests

File: README.md (hunk 2 above)
Comment (padamstx) [reply README.md]: in fact, it's required :)

File: Gopkg.toml (hunk 1 above)
Comment (padamstx) [reply Gopkg.toml]: We can probably declare the 1.0.0 release of the core when we deliver the actual major release.

File: README.md (hunk 3 above)
Comment (padamstx) [reply README.md]: in Go, each indent level is supposed to be a tab character

File: assistantv2/assistant_v2_test.go (hunk 4 above)
Comment (padamstx) [reply assistantv2/assistant_v2_test.go]: I made this change in my generator PR.

**Pull request merged**